x/feemarket/keeper/eip1559.go

```diff
@@ -56,13 +58,11 @@ func (k Keeper) CalculateBaseFee(ctx sdk.Context) *big.Int {
 
 	parentGasUsed := k.GetBlockGasWanted(ctx)
 
-	gasLimit := new(big.Int).SetUint64(math.MaxUint64)
-
 	// NOTE: a MaxGas equal to -1 means that block gas is unlimited
-	if consParams != nil && consParams.Block.MaxGas > -1 {
-		gasLimit = big.NewInt(consParams.Block.MaxGas)
+	if consParams == nil || consParams.Block == nil || consParams.Block.MaxGas <= -1 {
+		panic(fmt.Sprintf("get invalid consensus params: %s", consParams))
```

Comment:
Has this been backported? panic in the begin blocker case chain halt.
If for a reason we should use it, we should mention in comment the reasoning